### PR TITLE
*: drop go 1.13

### LINKED ISSUE #717: Use go 1.14 T.Cleanup() in tests
```
 The testing package now supports cleanup functions, called after a test or benchmark has finished, by calling T.Cleanup or B.Cleanup respectively. 
```
https://tip.golang.org/doc/go1.14

Very useful, we have some cases of tests writing to disk in `storage`, `compiler`, `wallet` and `io` packages (search for `os.RemoveAll` there) and it's a perfect fit for those.

Of course this could only be done when we're to require go 1.14 support (not now).

### LINKED ISSUE #1786: Drop Go 1.13 support
Remove it from builds/tests and solve #717.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #717, close #1786.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Builder image
-FROM golang:1-alpine as builder
+FROM golang:1.14-alpine as builder
```
Comment (roman-khimov) [inline Dockerfile:2]: Use latest, please.

File: pkg/core/notary_test.go [hunk 2]
```diff
@@ -661,6 +661,6 @@ func TestNotary(t *testing.T) {
 		mtx.RLock()
 		defer mtx.RUnlock()
 		return completedTxes[requests[0].MainTransaction.Hash()] != nil
-	}, time.Second, time.Millisecond)
+	}, 2*time.Second, 100*time.Millisecond)
```
Comment (roman-khimov) [inline pkg/core/notary_test.go:664]: s/100/10/?

File: Dockerfile (hunk 1 above)
Comment (AnnaShaleva) [reply Dockerfile]: Then let's leave it as is because `golang:1-alpine` is the latest.

**Review by roman-khimov — APPROVED** [review]

File: cli/executor_test.go [hunk 3]
```diff
@@ -102,6 +102,9 @@ func newExecutorWithConfig(t *testing.T, needChain bool, f func(*config.Config))
 	if needChain {
 		e.Chain, e.RPC, e.NetSrv = newTestChain(t, f)
 	}
+	t.Cleanup(func() {
```
Comment (fyrchik) [inline cli/executor_test.go:105]: Clean!

**Review by fyrchik — APPROVED** [review]

**Pull request merged**